packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingPlugin.kt

```diff
@@ -88,4 +96,91 @@ open class ExpoAutolinkingPlugin : Plugin<Project> {
       it.modules = modules
     }
   }
+
+  private fun findAppProject(root: Project, onFound: (Project) -> Unit): Project? {
+    val immediate = root.allprojects.firstOrNull { it.plugins.hasPlugin("com.android.application") }
+    if (immediate != null) {
+      onFound(immediate)
+      return immediate
+    }
+
+    root.allprojects.forEach { p ->
+      p.pluginManager.withPlugin("com.android.application") {
+        onFound(p)
+      }
+    }
+    return null
+  }
+
+  private fun copyAppDimensionsAndFlavorsToProject(
+    project: Project,
+    appProject: Project
+  ) {
+    val appAndroid = appProject.extensions.findByName("android") as? BaseExtension ?: run {
+      return
+    }
+    val consumerAndroid = project.extensions.findByName("android") as? BaseExtension ?: run {
+      return
+    }
+
+    val appDimensions = syncFlavorDimensions(project, consumerAndroid, appAndroid)
+    copyMissingProductFlavors(project, consumerAndroid, appAndroid, appDimensions)
+  }
+
+  private fun syncFlavorDimensions(
+    project: Project,
+    consumerAndroid: BaseExtension,
+    appAndroid: BaseExtension
+  ): List<String> {
+    val appDimensions = appAndroid.flavorDimensionList ?: emptyList()
+    if (appDimensions.isEmpty()) return emptyList()
+
+    val consumerDimensions = (consumerAndroid.flavorDimensionList ?: emptyList()).toMutableList()
+    val dimensionsAdded = appDimensions.any { dimension ->
+      if (dimension !in consumerDimensions) {
+        consumerDimensions.add(dimension)
+        true
+      } else {
+        false
+      }
+    }
+
+    if (dimensionsAdded || consumerAndroid.flavorDimensionList.isNullOrEmpty()) {
```

Comment:
```suggestion
    if (dimensionsAdded) {
```
Why do we need to check if `consumerAndroid.flavorDimensionList.isNullOrEmpty()`? Checking `dimensionsAdded` should be enough.